content/blog/2023-09-26-almalinux-jupyter.md

```diff
@@ -0,0 +1,52 @@
+date: '2023-09-26'
+post:
+  title: "Learn about how UCSB uses Almalinux in day-to-day operations."
+  image: /blog-images/23.09.26.ucsb_blog.svg
+---
+
+*Note: This post is a bit of a departure from what you normally see on the AlmaLinux blog, but we would love to share more stories like this! If you would like to write a post about how AlmaLinux helps you in your daily life, reach out to <a href="mailto:benny@almalinux.org">benny</a>.*
+
+The [College of Letters and Science at the University of California, Santa Barbara](https://lsit.ucsb.edu/) (UCSB) uses AlmaLinux, among other Linux distributions, in regular day-to-day operations. One such use case for AlmaLinux is the build infrastructure behind data science environments that serve hundreds to thousands of students per term.
+
+## Jupyter at UCSB
+
+Jupyter is used extensively at UCSB for classroom and research environments supporting Python and R language-driven research and curriculums, across such broad disciplines as data science, economics, linguistics, physics, psychology, and earth sciences. Traditionally, Jupyter and RStudio environments were either done in lab environments or on student machines. Unfortunately, this led to significant fragmentation of versioning and technical debt. Before long, machines were running old versions of things and were one well-meaning `pip install` or `install.packages()` away from fetching a library that was no longer compatible with older versions of the software. This means that student environments were easily broken, and TAs had to spend time fixing them and students sometimes had to learn on older, outdated versions of Python, R, and Jupyter.
+
+The solution to this was to move to [JupyerHub](https://jupyter.org/hub), deployed on Kubernetes, using [Zero To JupyterHub](https://zero-to-jupyterhub.readthedocs.io) as a resource for getting things started. Over time, the deployments themselves began to be standardized via Terraform with configs being tracked in git repos. Yet, while the deployments themselves became more manageable in this way, maintaining the images themselves was another problem. Often, each course had its own, often complex, requirements regarding libraries, extensions, and versions, and without a standardization on image bases. The technical debt of the lab environments ultimately found its way back in the form of container image sources that were now shipping the older versions. While still incrementally better (in that an updated Python library now keeps an image from being built rather than breaking a production lab environment or student machine's instance), this solution still meant potential for deploying vulnerabilities that were now being served over the public internet and didn't solve the problem of courses and research getting stuck on outdated versions.
+
+As more and more faculty turned to us to deploy Jupyter environments, the layers of complexity made things harder to maintain.
+
+## Continuous Integration
```

Comment:
`## Continuous Integration`

to 

`## Continuous Integration (CI)`

It is referenced as `CI` throughout the article. Maybe worth noting what it stands for here?